polaris/dataset/_base.py

```diff
@@ -0,0 +1,364 @@
+        than just indexing the `table` attribute. This method provides an end-point for seamlessly
+        accessing the underlying data.
+
+        Args:
+            row: The row index in the `Dataset.table` attribute
+            col: The column index in the `Dataset.table` attribute
+            adapters: The adapters to apply to the data before returning it.
+                If None, will use the default adapters specified for the dataset.
+
+        Returns:
+            A numpy array with the data at the specified indices. If the column is a pointer column,
+                the content of the referenced file is loaded to memory.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def upload_to_hub(self, access: AccessType = "private", owner: Union[HubOwner, str, None] = None):
+        """Uploads the dataset to the Polaris Hub."""
+        raise NotImplementedError
+
+    @classmethod
+    def from_json(cls, path: str):
+        """Loads a dataset from a JSON file.
+        Overrides the method from the base class to remove the caching dir from the file to load from,
+        as that should be user dependent.
+
+        Args:
+            path: Loads a dataset specification from a JSON file.
+        """
+        with fsspec.open(path, "r") as f:
+            data = json.load(f)
+        data.pop("cache_dir", None)
+        return cls.model_validate(data)
+
+    @abc.abstractmethod
+    def to_json(
+        self,
+        destination: str,
+        if_exists: ZarrConflictResolution = "replace",
+        load_zarr_from_new_location: bool = False,
+    ) -> str:
+        """
+        Save the dataset to a destination directory as a JSON file.
+
+        Args:
+            destination: The _directory_ to save the associated data to.
+            if_exists: Action for handling existing files in the Zarr archive. Options are 'raise' to throw
+                an error, 'replace' to overwrite, or 'skip' to proceed without altering the existing files.
+            load_zarr_from_new_location: Whether to update the current instance to load data from the location
+                the data is saved to. Only relevant for Zarr-datasets.
+
+        Returns:
+            The path to the JSON file.
+        """
+        raise NotImplementedError
+
+    def cache(self, verify_checksum: bool = True) -> str:
+        """Caches the dataset by downloading all additional data for pointer columns to a local directory.
+
+        Args:
+            verify_checksum: Whether to verify the checksum of the dataset after caching.
+
+        Returns:
+            The path to the cache directory.
+        """
+        self.to_json(self.cache_dir, load_zarr_from_new_location=True)
+
+        if verify_checksum:
+            self.verify_checksum()
+
+        return self.cache_dir
+
+    def size(self) -> tuple[int, int]:
+        return self.n_rows, self.n_columns
+
+    @abc.abstractmethod
+    def __getitem__(self, item):
```

Comment:
**Comment**
Since this isn't implemented in `DatasetV2`, it shouldn't be in the common parent class.